### PR TITLE
[KYUUBI #1323] Introduce EngineType to distinguish multiple engines

### PR DESCRIPTION
**Pull request by yanghua**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -758,6 +758,21 @@ object KyuubiConf {
     .version("1.2.0")
     .fallbackConf(LEGACY_ENGINE_SHARE_LEVEL)
 
+  val ENGINE_TYPE: ConfigEntry[String] = buildConf("engine.type")
+    .doc("Specify the detailed engine that supported by the Kyuubi. The engine type bindings to" +
+      " SESSION scope. Currently, available configs are: <ul>" +
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:763]: let's add `This configuration is experimental.`

Comment (yanghua) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:763]: OK~

**Review by pan3793 — APPROVED** [review]
LGTM

File: kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+package org.apache.kyuubi.engine
+
+/**
+ * Defines different engine types supported by Kyuubi.
+ */
+object EngineType extends Enumeration {
+  type EngineType = Value
+  val
+  SPARK_SQL,
+  FLINK_SQL = Value
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/engine/EngineType.scala:27]: Let's follow this scala style

* An enumeration class or object which extends the Enumeration class shall follow the convention for classes and objects, i.e. its name should be in PascalCase style. Enumeration values shall be in the upper case with words separated by the underscore character _. For example: 

```scala
  private object ParseState extends Enumeration {
  type ParseState = Value

  val PREFIX,
      TRIM_BEFORE_SIGN,
      SIGN,
      TRIM_BEFORE_VALUE,
      VALUE,
      VALUE_FRACTIONAL_PART,
      TRIM_BEFORE_UNIT,
      UNIT_BEGIN,
      UNIT_SUFFIX,
      UNIT_END = Value
}
```

**Review by yaooqinn — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Pull request closed without merging**